cql2pgjson/src/test/java/org/z3950/zing/cql/cql2pgjson/ForeignKeyGenerationTest.java

```diff
@@ -0,0 +1,126 @@
+package org.z3950.zing.cql.cql2pgjson;
+
+
+import static org.junit.Assert.*;
+import org.junit.runner.RunWith;
+
+import java.util.logging.Logger;
+
+import org.folio.cql2pgjson.CQL2PgJSON;
+import org.folio.cql2pgjson.exception.FieldException;
+import org.folio.cql2pgjson.exception.QueryValidationException;
+import org.folio.cql2pgjson.exception.ServerChoiceIndexesException;
+
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import junitparams.JUnitParamsRunner;
+
+
+
+@RunWith(JUnitParamsRunner.class)
+public class ForeignKeyGenerationTest  {
+  private static Logger logger = Logger.getLogger(CQL2PgJSONTest.class.getName());
+  private static CQL2PgJSON cql2pgJson;
+
+  @BeforeClass
+  public static void runOnceBeforeClass() throws Exception {
+  }
+  @AfterClass
+  public static void runOnceAfterClass() {
+  }
+  @Test
+  public void ForeignKeySearch() throws FieldException, QueryValidationException, ServerChoiceIndexesException {
+    CQL2PgJSON cql2pgJson = new CQL2PgJSON("tablea.json" );
+    cql2pgJson.setDbSchemaPath("templates/db_scripts/foreignKey.json");
+    String sql = cql2pgJson.toSql("tableb.tableb_data == 11111111-1111-1111-1111-111111111111").getWhere();
+    // default pkColumnName is id without underscore
+    assertEquals(" ( SELECT tableb.jsonb->>'tableb_data' from tableb WHERE (lower(f_unaccent(tablea.jsonb->>'id'::text)) = lower(f_unaccent(tableb.jsonb->>'tableaId'::text)))) LIKE lower(f_unaccent('11111111-1111-1111-1111-111111111111'))", sql);
+  }
+
+  @Test
+  public void ForeignKeySearchWithMissingFK() throws FieldException, QueryValidationException, ServerChoiceIndexesException {
+    CQL2PgJSON cql2pgJson = new CQL2PgJSON("tablea.json" );
+    cql2pgJson.setDbSchemaPath("templates/db_scripts/foreignKeyMissingFK.json");
+    try {
+      cql2pgJson.toSql("tableb.tableb_data == 11111111-1111-1111-1111-111111111111").getWhere();
+      fail();
+    } catch(QueryValidationException q) {
+      //left empty on purpose
+    }
+    
+  }
+  @Test
+  public void ForeignKeySearchWithMalformedFK() throws FieldException, QueryValidationException, ServerChoiceIndexesException {
+    CQL2PgJSON cql2pgJson = new CQL2PgJSON("tablea.json" );
+    cql2pgJson.setDbSchemaPath("templates/db_scripts/foreignKeyMalformedFK.json");
+    try {
+      cql2pgJson.toSql("tableb.tableb_data == 11111111-1111-1111-1111-111111111111").getWhere();
+      fail();
+    } catch(QueryValidationException q) {
+      //left empty on purpose
```

Comment:
Can you add a check that the exception message contains both "foreignKeys.targetTable missing" and the current table name?

You may use assertThrows from JUnit5 or ExpectedException from JUnit4.